File: template/nestJs/src/user/dto/update-user.dto.ts
import { PartialType } from '@nestjs/mapped-types';
import { CreateUserDto } from './create-user.dto';
import { IsNotEmpty } from 'class-validator';
import { i18nValidationMessage } from 'nestjs-i18n';
import { I18nTranslations } from '../../.generate/i18n.generated';
import { ApiProperty, ApiPropertyOptional } from '@nestjs/swagger';

export class UpdateUserDto extends PartialType(CreateUserDto) {
  @ApiPropertyOptional({
    description: '老密码'
  })
  oldPassword: string;
  @ApiPropertyOptional({
    description: '新密码'
  })
  newPassword: string;

  @ApiProperty({
    description: '电子邮箱'
  })
  @IsNotEmpty({
    message: i18nValidationMessage<I18nTranslations>('validation.NOT_EMPTY'),
  })
  email: string;
  @ApiProperty({
    description: '角色id',
    type: [Number]
  })
  roleIds: number[] = [];
  @ApiPropertyOptional()
  department: string;
  @ApiPropertyOptional()
  employeeType: string;
  @ApiPropertyOptional()
  probationStart: string;
  @ApiPropertyOptional()
  probationEnd: string;
  @ApiPropertyOptional()
  probationDuration: string;
  @ApiPropertyOptional()
  protocolStart: string;
  @ApiPropertyOptional()
  protocolEnd: string;
  @ApiPropertyOptional()
  address: string;
  @ApiPropertyOptional()
  status: number;
  @IsNotEmpty({
    message: i18nValidationMessage<I18nTranslations>('validation.NOT_EMPTY'),
  })
  name: string;
}

File: template/springboot/TinyPro/src/main/java/com/TinyPro/entity/dto/UpdateUserDto.java
package com.TinyPro.entity.dto;

import jakarta.validation.constraints.NotEmpty;
import lombok.Data;
import lombok.experimental.Accessors;

import java.util.ArrayList;
import java.util.List;

@Data
@Accessors(chain = true)
public class UpdateUserDto {
    
    @NotEmpty(message = "{NOT_EMPTY}")
    private String oldPassword;
    
    @NotEmpty(message = "{NOT_EMPTY}")
    private String newPassword;
    
    @NotEmpty(message = "{NOT_EMPTY}")
    private String email;
    
    @NotEmpty(message = "{NOT_EMPTY}")
    private List<Long> roleIds = new ArrayList<>();
    
    @NotEmpty(message = "{NOT_EMPTY}")
    private String department;
    
    @NotEmpty(message = "{NOT_EMPTY}")
    private String employeeType;
    
    @NotEmpty(message = "{NOT_EMPTY}")
    private String probationStart;
    
    @NotEmpty(message = "{NOT_EMPTY}")
    private String probationEnd;
    
    @NotEmpty(message = "{NOT_EMPTY}")
    private String probationDuration;
    
    @NotEmpty(message = "{NOT_EMPTY}")
    private String protocolStart;
    
    @NotEmpty(message = "{NOT_EMPTY}")
    private String protocolEnd;
    
    @NotEmpty(message = "{NOT_EMPTY}")
    private String address;
    
    @NotEmpty(message = "{NOT_EMPTY}")
    private Integer status;
    
    @NotEmpty(message = "{NOT_EMPTY}")
    private String name;
}